src/poetry/installation/installer.py

```diff
@@ -53,7 +53,6 @@ def __init__(
         self._requires_synchronization = False
         self._update = False
         self._verbose = False
-        self._write_lock = True
```

Comment:
Another unnecessary state variable goes away, as basically `_write_lock = not dry_run`

Reply:
Nice. 👍